src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +241,501 @@ otError DuaManager::Store(void)
+            break;
+        case ThreadStatusTlv::kDuaInvalid:
+        case ThreadStatusTlv::kDuaDuplicate:
+            SendAddressNotification(target, static_cast<ThreadStatusTlv::DuaStatus>(status), *child);
+            IgnoreError(child->RemoveIp6Address(target));
+            mChildDuaMask.Clear(mChildIndexDuaRegistering);
+            mChildDuaRegisteredMask.Clear(mChildIndexDuaRegistering);
+            break;
+        case ThreadStatusTlv::kDuaNoResources:
+        case ThreadStatusTlv::kDuaNotPrimary:
+        case ThreadStatusTlv::kDuaGeneralFailure:
+            UpdateReregistrationDelay();
+            break;
+        }
+    }
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+
+exit:
+    ScheduleTimer();
+    return error;
+}
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+void DuaManager::SendAddressNotification(Ip6::Address &             aAddress,
+                                         ThreadStatusTlv::DuaStatus aStatus,
+                                         const Child &              aChild)
+{
+    Coap::Message *  message = nullptr;
+    Ip6::MessageInfo messageInfo;
+    otError          error;
+
+    VerifyOrExit((message = Get<Coap::Coap>().NewPriorityMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error =
+                      message->Init(OT_COAP_TYPE_CONFIRMABLE, OT_COAP_CODE_POST, OT_URI_PATH_DUA_REGISTRATION_NOTIFY));
+    SuccessOrExit(error = message->SetPayloadMarker());
+
+    SuccessOrExit(error = Tlv::AppendUint8Tlv(*message, ThreadTlv::kStatus, static_cast<uint8_t>(aStatus)));
+    SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &aAddress, sizeof(aAddress)));
+
+    messageInfo.GetPeerAddr().SetToRoutingLocator(Get<Mle::MleRouter>().GetMeshLocalPrefix(), aChild.GetRloc16());
+    messageInfo.SetPeerPort(kCoapUdpPort);
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+
+    SuccessOrExit(error = Get<Coap::Coap>().SendMessage(*message, messageInfo));
+
+    otLogInfoDua("Sent ADDR_NTF for child %04x DUA %s", aChild.GetRloc16(), aAddress.ToString().AsCString());
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        // TODO: (DUA) (P4) may enhance to  guarantee the delivery of DUA.ntf
+        otLogWarnDua("Sent ADDR_NTF for child %04x DUA %s Error %s", aChild.GetRloc16(),
+                     aAddress.ToString().AsCString(), otThreadErrorToString(error));
+
+        if (message != NULL)
+        {
+            message->Free();
+        }
+    }
+}
+
+void DuaManager::UpdateChildDomainUnicastAddress(const Child &aChild, Mle::ChildDuaState aState)
+{
+    uint16_t childIndex = Get<ChildTable>().GetChildIndex(aChild);
+
+    if (aState == Mle::ChildDuaState::kRemoved || aState == Mle::ChildDuaState::kChanged)
+    {
+        VerifyOrExit(mChildDuaMask.Get(childIndex) == true, OT_NOOP);
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+        if (mIsDuaPending && mDuaState != DuaState::kRegistering && mChildIndexDuaRegistering == childIndex)
```

Comment:
```suggestion
        if (mIsDuaPending && (mDuaState != DuaState::kRegistering || mChildIndexDuaRegistering == childIndex))
```

Reply:
why change to `||`? 
Only need to AbortTransaction is the FTD is registering for this child when the child's DUA changes, right?